### DISCUSSION TITLE
Preview thumbnail for new scenario screen

### DISCUSSION THREAD

**Discussion by GingerAdonis** [OpenRCT2/OpenRCT2#15680] [Feature requests] [answered]
It would be a nice addition to have a preview thumbnail of the map before starting a new scenario.

Posted on behalf fuyangli ( https://www.reddit.com/r/openrct2/comments/4cqrco/request_feature/ )

**Comment by AaronVanGeffen** [comment]
Once we're moving to our own scenario file format, this would be an easy feature to add. Until then, the performance penalty for such a feature is too much of a downside, I think.

**Comment by GingerAdonis** [comment]
Isn't it something we can pre-render for the default scenarios, as a temporary work-around? Or render all scenarios the user has installed on first run.

**Comment by AaronVanGeffen** [comment]
And cache them in a directory in the user settings folder? Sure, but I'd rather not, to be honest — explain to the average user why it's taking minutes to launch OpenRCT2 for the first time.

If it's an opt-in feature, then sure — but that probably defeats its purpose, and the amount of time it takes to make the feature in the first place. As such, I'd rather we defer.

**Comment by ghost** [comment]
This feature reminds me of Chris Sawyer's Locomotion when you hover over a scenario name when starting a new game, you'll see what the scenario looks like in map form.

**Comment by GingerAdonis** [comment]
I read more and more the project needs a custom save game format. What is stopping us from doing that? Backwards compatibility is a good reason, but it'll have to happen at one point. (this is a more general question)

**Comment by IntelOrca** [comment]
@JarnoVgr Not enough players are ready to make that one-way leap yet. We want to have completed everything first, refactored all the code so that we can then carefully design a new superior park format and code it well.

And you'll find a lot of things still can't be improved with a new format alone. Its the remaining RCT2 code that restricts as (all the drawing code).

**Comment by Krutonium** [comment]
(I really want json)

**Comment by Superjustinbros** [comment]
Would it be possible for something similar to be implemented into save files as well? I'm guessing not because of how much more stuff is usually saved into those compared to scenario files.

**Comment by janisozaur** [comment]
See https://github.com/OpenRCT2/OpenRCT2/pull/2348

**Comment by fuyangli** [comment]
What about a simple png corresponding to that scenario? I mean, pre-rendered of the map to show how the map is. And then, after the full reimplementation of the code, work in some render with the save game!
(Btw: I am a programmer, and I want to collaborate with the code. How can I do that?) :D

**Comment by Krutonium** [comment]
@fuyangli Jump on Gitter if you havn't yet and ask :P

**Comment by Broxzier** [comment]
janisozaur's PR about PNG embesaves was the first thing I thought of too. I can see this work for when you save a game, it would also export a screenshot or thumbnail of the park within the saved game file.

@fuyangli: https://gitter.im/OpenRCT2/OpenRCT2

**Comment by GingerAdonis** [comment]
What @fuyangli says shouldn't be too hard to develop and is backwards compatible. When embedding saves into a PNG the backwards compatibility is gone.

**Comment by Gymnasiast** [comment]
Even with @fuyangli 's plan, @AaronVanGeffen 's point still stands. Please be a little bit more patient so we can do it properly.

**Comment by GingerAdonis** [comment]
@Gymnasiast What performance penalty is there? The only thing I can think of is wait time when saving because of the giant screenshot that needs to be made. Which is a bad thing indeed. But is that performance hit also big when rendering a smaller/thumbnail giant screenshot?

**Comment by Gymnasiast** [comment]
It would have to either pre-generate a thumbnail for every scenario (making the first-time load take several minutes) or generate one on the fly, making scrolling through the list sluggish. We can design our own format to have a provision for a built-in thumbnail.

**Comment by GingerAdonis** [comment]
Well I was now talking more about saved games. But for scenarios we could include these into the packaging for retail scenarios. Even if we would have a new scenario format, it would still need to be generated at first run for SC6 scenarios, right? And that will happen at some point anyways, I bet.

**Comment by Gymnasiast** [comment]
Hm, I see. I'm not sure if it's worth adding support for thumbs now, considering we'll have to replace it again when we do move. But if someone can add support in a way that doesn't have bad side effects and isn't intrusive it will probably get merged.

**Comment by GingerAdonis** [comment]
But the thing is, it won't need to be replaced. It would only need to add support for new save files/scenario formats, which would make it more efficient, but the old mechanics will probably need to stay there because of backwards compatibility. I don't see OpenRCT2 dropping support for loading SV6 and SC6 files anytime soon. But we'll see if someone wants to code this. I would personally like to see this happen.

**Comment by Gymnasiast** [comment]
I expect OpenRCT2 to import SC6/SV6/SC4/SV4 to its own format first, as having to add support for three types of file in the scenario selector makes the code a lot dirtier. So it would have to be replaced if we indeed go down that route.

**Comment by tellovishous** [comment]
Bumping this since the new save format has been implemented. Sounds like a nice feature, maybe this could be looked into now?

**Comment by J-PIE-314** [comment]
@AaronVanGeffen is planning to implement this feature in PR #22646.

**Comment by ZeeMaji** [comment, marked as answer]
This has been implemented for all official scenarios and any .park scenarios made since i believe #23876 with an option to use either screenshots or maps. Custom sc4/sc6 or old .park scenarios still lack previews but i don't think anything can be done about that.
